docs/ldml/tr35-keyboards.md

```diff
@@ -1378,30 +1380,149 @@ This is a container for variables to be used with transforms.
+> [Markers](#markers) may be included with the `\m{…}` notation.
+
+**Examples**
+
+```xml
+<variables>
+    <set id="upper" value="A B CC D E FF " /> <!-- 6 items -->
+    <set id="lower" value="a b c  d e  f " /> <!-- 6 items -->
+    <set id="upper_or_lower" value="$[upper] $[lower]"  /> <!-- Concatenate two sets -->
+    <set id="lower_or_upper" value="$[lower] $[upper]"  /> <!-- Concatenate two sets -->
+    <set id="a" value="A"/> <!-- Just one element, an 'A'-->
+    <set id="cluster_or_zwnj" value="${hi_cluster} ${zwnj}"/> <!-- 2 items: "हि \u${200C}"-->
+</variables>
+```
+
+Match "X" followed by any uppercase letter:
+
+```xml
+<transform from="X$[upper]" to="…" />
+```
+
+Map from upper to lower:
 
+```xml
+<transform from="($[upper])" to="$[1:lower]" />
 ```
 
-See the `from=` and `to=` sections of [`transforms`](#Element_transforms) for details.
+See [transform](#element-transform) for further details.
+
+* * *
+
+### Element: unicodeSet
+
+> <small>
+>
+> Parents: [variables](#Element_variables)
+>
+> Children: _none_
+>
+> Occurrence: optional, multiple
+> </small>
+
+> This element represents a set, using a subset of the [UnicodeSet](tr35.md#Unicode_Sets) format, used by the [transform](#element-transform) elements for string matching and substitution.
+> Note important restrictions on the syntax below.
+
+_Attribute:_ `id` (required)
+
+> Specifies the identifier (name) of this unicodeSet.
+> All ids must be unique across all types of variables.
+
+_Attribute:_ `value` (required)
+
+> String value in [UnicodeSet](tr35.md#Unicode_Sets) format.
+> Leading and trailing whitespace is ignored.
+> Variables may refer to other string variables if they have been previously defined, using `${string}` syntax, or to other previously-defined UnicodeSets (not sets) using `$[unicodeSet]` syntax.
+
+**Syntax Note**
+
+- Multi-character strings (`{}`) are not supported, such as `[żġħ{ie}{għ}]`.
+- UnicodeSet property notation (`\p{…}` or `[:…:]`) may **NOT** be used, because that would make implementations dependent on a particular version of Unicode. However, implementations and tools may wish to pre-calculate the value of a particular UnicodeSet, and "freeze" it as explicit code points.  The example below of `$[KhmrMn]` matches all nonspacing marks in the `Khmr` script.
+- UnicodeSets may represent a very large number of codepoints. A limit may be set on how many unique range entries may be matched.
+
+**Examples**
+
+```xml
+<variables>
+<unicodeSet id="consonants" value="[कसतनमह]" /> <!-- unicode set range -->
+<unicodeSet id="range" value="[a-z D E F G \u200A]" /> <!-- a through z, plus a few others -->
+<unicodeSet id="newrange" value="[$[range]-[G]]" /> <!-- The above range, but not including G -->
+<unicodeSet id="KhmrMn" value="[[\u17B4\u17B5\u17B7-\u17BD\u17C6\u17C9-\u17D3\u17DD]"> <!--  [[:Khmr:][:Mn:]] as of Unicode 15.0-->
+</variables>
+```
+
+The `unicodeSet` element may not be used in mapping operations.
```

Comment:
Perhaps we need to say "may not be referenced by key and display elements"?
```suggestion
The `unicodeSet` element may not be referenced by [`key`](#element-key) and [`display`](#element-display) elements.
```

Reply:
It can't be used in those, either.   I think that should be separate.

Reply:
```suggestion
The `unicodeSet` element may not be used as the source or target for mapping operations (`$[1:variable]` syntax), nor referenced by [`key`](#element-key) and [`display`](#element-display) elements.
```